_includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java

```diff
@@ -0,0 +1,171 @@
+package io.confluent.developer;
+
+
+import com.google.common.io.Resources;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.flink.api.common.restartstrategy.RestartStrategies;
+import org.apache.flink.contrib.streaming.state.EmbeddedRocksDBStateBackend;
+import org.apache.flink.runtime.client.JobCancellationException;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.assertj.core.util.Sets;
+import org.junit.Before;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.testcontainers.containers.KafkaContainer.KAFKA_PORT;
+
+/**
+ * Base class for Flink SQL integration tests that use Flink's Kafka connectors. Encapsulates
+ * Kafka broker and Schema Registry Testcontainer management and includes utility methods for
+ * dynamically configuring Flink SQL Kafka connectors and processing Table API results.
+ */
+public class AbstractFlinkKafkaTest {
+
+  protected StreamTableEnvironment tEnv;
```

Comment:
```suggestion
  protected StreamTableEnvironment streamTableEnv;
```
Super nit - feel free to ignore

Reply:
agree, changed